src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
+
+        }
+    }
+
+    public MyHashMap() {
+
+    }
 
     @Override
     public void put(K key, V value) {
+        int arrayIndex = arrayIndexCalculation(key);
+        MyNode<K,V> tempNode = new MyNode<>();
+        MyNode<K,V> newNode = new MyNode<>(key != null ? key.hashCode() : 0, key, value, null);
+        tempNode = bucketArray[arrayIndex];
 
+        do {
+            if (tempNode == null) {
+                bucketArray[arrayIndex] = newNode;
+                size++;
+                break;
+            } else if (newNode.key == null && tempNode.key == newNode.key) {
+                tempNode.value = newNode.value;
+                break;
+            } else if (newNode.key == null && tempNode.next == null) {
+                tempNode.next = newNode;
+                size++;
+                break;
+            } else if (tempNode.key != null && tempNode.key.equals(newNode.key)) {
+                tempNode.value = newNode.value;
+                break;
+            } else if (tempNode.next == null) {
+                tempNode.next = newNode;
+                size++;
+                break;
+            }
+            tempNode = tempNode.next;
+        } while (tempNode != null);
+        if (size == bucketArray.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        return getRemove(key, true);
+    }
+
+    public void clear() {
+        bucketArray = new MyNode[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    public V remove(K key) {
+        return getRemove(key, false);
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private V getRemove(K key, boolean getTrue) {
+        int arrayIndex = arrayIndexCalculation(key);
+        MyNode<K, V> tempNode = new MyNode<>();
+        tempNode = bucketArray[arrayIndex];
+        if (tempNode != null) {
+            do {
+                if ((tempNode.key == null && key == null)
+                        || (key != null && key.equals(tempNode.key))) {
+                    if (!getTrue) {
+                        tempNode.next = tempNode.next.next;
+                        size--;
+                    }
+                    return tempNode.value;
+                } else {
+                    tempNode = tempNode.next;
+                }
+            } while (tempNode != null);
+        }
+
+        return null;
+    }
+
+    private MyNode<K, V>[] resize() {
+        MyNode<K,V>[] oldBucketArray = new MyNode[bucketArray.length];
+
+        oldBucketArray = bucketArray;
+        int oldSize = size;
+        size = 0;
+        bucketArray = new MyNode[bucketArray.length << 1];
+        for (MyNode<K,V> tempNode : oldBucketArray) {
+            do {
+                if (tempNode != null) {
+                    put(tempNode.key, tempNode.value);
+                    tempNode = tempNode.next;
+                }
+            } while (tempNode != null);
```

Comment:
```suggestion
            while (tempNode != null) {
                    put(tempNode.key, tempNode.value);
                    tempNode = tempNode.next;
            } 
```
Think about using while loop everywhere instead of do-while in this task.